### PR TITLE
Add British Columbia Canada Provider

### PR DESCRIPTION
**Pull request by anjandev**
resolving: https://github.com/grote/Transportr/issues/434

### CODE REVIEW

**Comment by anjandev** [comment]
The app crashes when I search for "nearby stations" and a station nearby is a skytrain station. The app does not crash if nearby stations are only bus stations. I can provide a log when I get home. I was thinking that this could be fixed by setting
 
```kotlin
goodLineNames = true,
```
to:
```kotlin
goodLineNames = false,
```

However, I do not know what this variable does.

**Comment by anjandev** [comment]
```05-11 23:07:13.681 16374-16374/de.grobox.liberario.debug E/AndroidRuntime: FATAL EXCEPTION: main
    Process: de.grobox.liberario.debug, PID: 16374
    java.util.NoSuchElementException
        at java.util.RegularEnumSet$EnumSetIterator.next(RegularEnumSet.java:101)
        at java.util.RegularEnumSet$EnumSetIterator.next(RegularEnumSet.java:99)
        at de.grobox.transportr.map.NearbyStationsDrawer.getIconForProduct(NearbyStationsDrawer.kt:65)
        at de.grobox.transportr.map.NearbyStationsDrawer.draw(NearbyStationsDrawer.kt:44)
        at de.grobox.transportr.map.MapFragment.onLoadFinished(MapFragment.kt:173)
        at de.grobox.transportr.map.MapFragment.onLoadFinished(MapFragment.kt:49)
        at android.support.v4.app.LoaderManagerImpl$LoaderObserver.onChanged(LoaderManagerImpl.java:221)
        at android.arch.lifecycle.LiveData.considerNotify(LiveData.java:109)
        at android.arch.lifecycle.LiveData.dispatchingValue(LiveData.java:126)
        at android.arch.lifecycle.LiveData.setValue(LiveData.java:282)
        at android.arch.lifecycle.MutableLiveData.setValue(MutableLiveData.java:33)
        at android.support.v4.app.LoaderManagerImpl$LoaderInfo.onLoadComplete(LoaderManagerImpl.java:156)
        at android.support.v4.content.Loader.deliverResult(Loader.java:130)
        at android.support.v4.content.AsyncTaskLoader.dispatchOnLoadComplete(AsyncTaskLoader.java:257)
        at android.support.v4.content.AsyncTaskLoader$LoadTask.onPostExecute(AsyncTaskLoader.java:82)
        at android.support.v4.content.ModernAsyncTask.finish(ModernAsyncTask.java:487)
        at android.support.v4.content.ModernAsyncTask$InternalHandler.handleMessage(ModernAsyncTask.java:504)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6186)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
```

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -257,6 +257,9 @@ and always knows where you are to not miss where to get off the bus.
 	<string name="np_name_quebec">Quebec</string>
 	<string name="np_desc_quebec">Deux-Montagnes, Laval, L\'Assomption, Outaouais, Sud-Ouest, Quebec, Haut-Saint-Laurent, Lanaudière, La Presqu\'Île, Laurentides, Montreal, Les Moulins, Vallée du Richelieu, Chambly-Richelieu-Carignan, Roussillon, Sorel-Varennes, Le Richelain, Sherbrooke, Sainte-Julie</string>
 	<string name="np_desc_quebec_networks" translatable="false">AMT, STM, STL</string>
+	<string name="np_name_britishcolumbia">British Columbia</string>
+	<string name="np_desc_britishcolumbia">Public transportation in BC</string>
+	<string name="np_desc_britishcolumbia_networks" translatable="false">Translink (Metro Vancouver) and BC Transit (rest of BC)</string>
```
[the remark this answers is not in the document]
Comment (anjandev) [reply app/src/main/res/values/strings.xml]: That was a mistake. Will fix

**Comment by grote** [comment]
Looks like this here is trying to get an icon for a missing (empty?) product?
```
        at de.grobox.transportr.map.NearbyStationsDrawer.getIconForProduct(NearbyStationsDrawer.kt:65)
        at de.grobox.transportr.map.NearbyStationsDrawer.draw(NearbyStationsDrawer.kt:44)
        at de.grobox.transportr.map.MapFragment.onLoadFinished(MapFragment.kt:173)
```
Could be a Transportr bug.

**Comment by grote** [comment]
Looks like you found indeed a Transportr bug. Thanks! Should be fixed with e8ab77d825b1d44e93a457e63e432384c844034b. Could you please rebase on latest master and try again?

**Comment by anjandev** [comment]
@grote e8ab77d seems to have resolved the issue.

**Pull request merged**